### PR TITLE
FIX Issue recursion when parent/child records are swapped in SiteTree

### PR DESCRIPTION
**Pull request by dhensby**
see #776

### CODE REVIEW

**Comment by dhensby** [comment]
Don't merge

File: code/model/SiteTree.php [hunk 1]
```diff
@@ -1158,10 +1158,11 @@ static public function prepopulate_permission_cache($permission = 'CanEditType',
 	 * @param string $siteConfigMethod Method to call on {@link SiteConfig} for toplevel items, e.g. "canEdit"
 	 * @param string $globalPermission If the member doesn't have this permission code, don't bother iterating deeper
 	 * @param bool   $useCached
+	 * @param array $stages            Which stages to check permissions against, defaults to both Stage and Live
 	 * @return array An map of {@link SiteTree} ID keys to boolean values
 	 */
 	public static function batch_permission_check($ids, $memberID, $typeField, $groupJoinTable, $siteConfigMethod,
-												  $globalPermission = null, $useCached = true) {
+												  $globalPermission = null, $useCached = true, $stages = array('Stage', 'Live')) {
```
Comment (tractorcow) [inline code/model/SiteTree.php]: If you add $stages you'll need to add this to the cache key.

Comment (dhensby) [inline code/model/SiteTree.php]: done

**Comment by frankmullenger** [comment]
@dhensby @tractorcow many thanks!

**Pull request merged**